### PR TITLE
Enable Panache-style enhancement for public fields for all Hibernate entities, even without Panache 

### LINKED ISSUE #12873: Hibernate does not perform an update statement without setters
**Describe the bug**

If an entity (see example below) is attached to the persistence context (or a merge is called manually) and the application tries to update a property AND there are not setters for this property (or in general), the update is not perfomed. 
See following discussion with @Sanne https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/hibernate.20(merge).20with.20public.20fields

```
@Entity
public class Event  {

  @NotNull
  public String name;

  @Id
  @GeneratedValue
  public Long id;
}
```
Having the above entity and trying to perform an update to the `name` with 
```
@PATCH
  @Transactional
  public void appendToGreeting() {
    entityManager.find(Event.class, 1L).name = "public";
  }
```

**Expected behavior**
It is expected to perform an update statement to the underlying database, if the name has changed (and if it stays the same, I am not sure here). 

**Actual behavior**
No update statement is performed and therefor the value has not changed. 

**To Reproduce**

Add the code above to a plain quarkus project with hibernate-orm as dependency

```
unname: Linux
Quarkus version: 1.8.1.Final
Java: 
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-8u265-b01-0ubuntu2~18.04-b01)
OpenJDK 64-Bit Server VM (build 25.265-b01, mixed mode)
Build Tool: Maven

Without native mode! 


Reproducer:
https://github.com/ahofmeister/quarkus-orm-reproducer

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #12873

As discussed with @Sanne .

Initiallly we considered just using the "extended enhancement" that Hibernate ORM bundles with, but the main concern was that performance wouldn't be great as that enhancement is performed at a higher level (Bytebuddy) which makes it hard to integrate into Quarkus. We would have had to basically load every class to convert as a `byte[]`, transform it, then write it, pretty much like we do in `io.quarkus.hibernate.orm.deployment.HibernateEntityEnhancer`, but... for **every class in the application**, since every class can potentially access public fields of entities.

In the end we agreed on a different approach: taking Panache's field access enhancement and making it generic enough to be used even when panache is not enabled. The main advantage is that Panache's field access enhancement is already tightly integrated with Quarkus' bytecode enhancement.

In practice, this means some code had to be split in Panache to clearly separate what has to do with field access enhancement from what has to do with Panache's generation of Active-Record-style static methods. Then all I had to do was to add a dependency from the Hibernate ORM extension to Panache's *common* extension (which doesn't do anything by itself), and wire everything so that all Hibernate ORM entities get field-access enhanced, not just Panache entities.

### CODE REVIEW

**Comment by gavinking** [comment]
Does this also apply to HR entities?

**Comment by yrodiere** [comment]
> Does this also apply to HR entities?

It does. See:

* [`extensions/hibernate-reactive/deployment/src/test/java/io/quarkus/hibernate/reactive/publicfields/PublicFieldAccessFieldTypesTest.java`](https://github.com/quarkusio/quarkus/pull/15230/files#diff-4cf233bbcbf15d12b425eecb709886753f6f6541347cc1fc7292e57b266328f7)
* [`extensions/hibernate-reactive/deployment/src/test/java/io/quarkus/hibernate/reactive/publicfields/PublicFieldAccessInheritanceTest.java`](https://github.com/quarkusio/quarkus/pull/15230/files#diff-cffcfe32d3f84f5b55f959f441047b500aa813b36bd10356215cc9e91e23494a)

**Comment by gavinking** [comment]
Excellent! That's great.

File: extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ private void processEntities(CombinedIndexBuildItem index,
         for (Map.Entry<String, Set<String>> entry : collectedEntityToPersistenceUnits.entrySet()) {
             String entityName = entry.getKey();
             List<String> selectedPersistenceUnits = new ArrayList<>(entry.getValue());
-            boolean isPanacheEntity = modelClasses.stream().anyMatch(name -> name.equals(entityName));
```
Comment (Sanne) [inline extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java:154]: funny code :)

File: extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java [hunk 2]
```diff
@@ -59,12 +59,11 @@
 
     @BuildStep
     @Record(ExecutionTime.STATIC_INIT)
+    @Consume(HibernateEnhancersRegisteredBuildItem.class)
```
Comment (Sanne) [inline extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java:60]: I had no idea about this `Consume` annotation, thanks.

**Review by Sanne — APPROVED** [review]
Fantastic job, thank you so much.

**Pull request merged**